eng/common/TestResources/New-TestResources.ps1

```diff
@@ -279,46 +279,13 @@ try {
         if (!$TenantId) {
             $TenantId = $context.Subscription.TenantId
         }
-
-        # If no test application ID is specified during an interactive session, create a new service principal.
-        if (!$TestApplicationId) {
-
-            # Cache the created service principal in this session for frequent reuse.
-            $servicePrincipal = if ($AzureTestPrincipal -and (Get-AzADServicePrincipal -ApplicationId $AzureTestPrincipal.ApplicationId) -and $AzureTestSubscription -eq $SubscriptionId) {
-                Log "TestApplicationId was not specified; loading cached service principal '$($AzureTestPrincipal.ApplicationId)'"
-                $AzureTestPrincipal
-            } else {
-                Log "TestApplicationId was not specified; creating a new service principal in subscription '$SubscriptionId'"
-                $suffix = (New-Guid).ToString('n').Substring(0, 4)
-                $global:AzureTestPrincipal = New-AzADServicePrincipal -Role Owner -Scope "/subscriptions/$SubscriptionId" -DisplayName "test-resources-$($baseName)$suffix.microsoft.com"
-                $global:AzureTestSubscription = $SubscriptionId
-
-                Log "Created service principal '$($AzureTestPrincipal.ApplicationId)'"
-                $AzureTestPrincipal
-            }
-
-            $TestApplicationId = $servicePrincipal.ApplicationId
-            $TestApplicationSecret = (ConvertFrom-SecureString $servicePrincipal.Secret -AsPlainText);
-
-            # Make sure pre- and post-scripts are passed formerly required arguments.
-            $PSBoundParameters['TestApplicationId'] = $TestApplicationId
-            $PSBoundParameters['TestApplicationSecret'] = $TestApplicationSecret
-        }
-
-        if (!$ProvisionerApplicationId) {
-            $ProvisionerApplicationId = $TestApplicationId
-            $ProvisionerApplicationSecret = $TestApplicationSecret
-            $TenantId = $context.Tenant.Id
-        }
     }
 
-    # Log in as and run pre- and post-scripts as the provisioner service principal.
+    # If a provisioner service principal was provided, log into it to perform the pre- and post-scripts and deployments.
     if ($ProvisionerApplicationId) {
```

Comment:
The provisioner has to be set the test application if not provided directly, which we don't expect customers to do. This is why all that work above you deleted is done prior to this line.

Reply:
Why would that be necessary?  I believe that it's inappropriate for the Test Application to be used as the provisioner.   The script should be running in the context of the caller if no provisioner is provided so that the Test Application can be restricted to the RG and not require rights to the subscription for creating resources.

Reply:
Because the user might not be logged in. At the very least we now need to mandate they do log in first. I'm not opposed to it, but it's an extra manual step. The point of these scripts was to automate as much as possible.

Reply:
Again, I may be overlooking something... but how can you create an AAD service principal if you're not logged in?   If I call `New-AzADServicePrincipal` without being logged in,  it fails and I'm prompted to login.      

If I run the current `New-TestResources` without being logged in, I'm prompted to login (line 228).    That experience would be the same with these changes, no?

Reply:
Discussed offline.